manta-trusted-setup/src/groth16/ceremony/config/ppot.rs

```diff
@@ -308,30 +325,65 @@ pub fn register(twitter_account: String, email: String) {
 
 /// Prompts the client information and get client keys.
 #[inline]
-pub fn get_client_keys() -> Option<(ed25519::SecretKey, ed25519::PublicKey)> {
+pub fn get_client_keys() -> Result<(ed25519::SecretKey, ed25519::PublicKey), ClientKeyError> {
     println!(
         "Please enter the {} you received when you registered yourself using this tool.",
         "Secret".italic()
     );
-    let seed_bytes = Seed::new(
-        &Mnemonic::from_phrase(
-            Input::with_theme(&ColorfulTheme::default())
-                .with_prompt("Your Secret")
-                .validate_with(|input: &String| -> Result<(), &str> {
-                    Mnemonic::validate(input, Language::English)
-                        .map_err(|_| "This is not a valid secret.")
-                })
-                .interact_text()
-                .expect("Please enter your secret received during `Register`.")
-                .as_str(),
-            Language::English,
-        )
-        .expect("Should produce a mnemonic from the secret."),
-        "manta-trusted-setup",
-    )
-    .as_bytes()
-    .to_vec();
-    generate_keys(&seed_bytes)
+    let text = match Input::with_theme(&ColorfulTheme::default())
+        .with_prompt("Your Secret")
+        .validate_with(|input: &String| -> Result<(), &str> {
+            Mnemonic::validate(input, Language::English).map_err(|_| "This is not a valid secret.")
+        })
+        .interact_text()
+    {
+        Ok(text) => text,
+        Err(_) => return Err(ClientKeyError::InvalidSecret),
+    };
+    let mnemonic = match Mnemonic::from_phrase(text.as_str(), Language::English) {
+        Ok(mnemonic) => mnemonic,
+        Err(_) => return Err(ClientKeyError::MnemonicFailure),
+    };
+    let seed_bytes = Seed::new(&mnemonic, "manta-trusted-setup")
+        .as_bytes()
+        .to_vec();
+    match generate_keys(&seed_bytes) {
+        Some(keys) => Ok(keys),
+        None => Err(ClientKeyError::KeyGenerationFailure),
+    }
+}
+
+/// Client Key Error
+pub enum ClientKeyError {
```

Comment:
Make sure to derive all the other standard traits when possible:

```suggestion
#[derive(Clone, Copy, Debug, Deserialize, Eq, Hash, PartialEq, Serialize)]
#[serde(crate = "manta_util::serde", deny_unknown_fields)]
pub enum ClientKeyError {
```